### PR TITLE
[RFC] Lift subdirectory restriction on copy_files stanza?

### PR DESCRIPTION
**Pull request by nojb**
In https://github.com/ocaml/dune/issues/911#issuecomment-399604375, it is implied that this restriction should not be required anymore.

I don't yet know enough about the inner workings of `dune` to investigate this on my own, so:

- Can this restriction indeed be removed?
- If yes, is there anything else to adapt to make it so? What is the `merlin` issue alluded to in the comment in the code?

### CODE REVIEW

**Comment by nojb** [comment]
Thanks for the review! I pushed a commit enabling this only from 1.3 onwards.

**Comment by rgrinberg** [comment]
Would you mind updating the `copy_files` tests? You might need to organize them a little better with different roots for some test cases (you can look at the `multi-dir`  tests for an example of how to do that).

Also, don't forget a CHANGES entry.

**Comment by nojb** [comment]
Updated CHANGES and added a small test.

File: src/simple_rules.ml [hunk 1]
```diff
@@ -66,11 +66,7 @@ let copy_files sctx ~dir ~scope ~src_dir (def: Copy_files.t) =
     let src_glob = SC.expand_vars_string sctx ~dir def.glob ~scope in
     Path.relative src_dir src_glob ~error_loc:loc
   in
-  (* The following condition is required for merlin to work.
-     Additionally, the order in which the rules are evaluated only
-     ensures that [sources_and_targets_known_so_far] returns the
-     right answer for sub-directories only. *)
-  if not (Path.is_descendant glob_in_src ~of_:src_dir) then
+  if def.syntax_version < (1, 3) && not (Path.is_descendant glob_in_src ~of_:src_dir) then
```
Comment (rgrinberg) [inline src/simple_rules.ml]: Just a note that we usually wrap our code to 80 chars. I'm not sure if it's over the limit here, but something to keep in mind.

**Review by rgrinberg — APPROVED** [review]
Looks good nojb. Thanks for the contribution

**Review by emillon — APPROVED** [review]

**Pull request merged**